src/main/java/core/basesyntax/utils/transaction/handlers/OperationHandler.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.utils.transaction.handlers;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+
+public interface OperationHandler {
+    void perform(FruitTransaction transaction, StorageImpl storage);
```

Comment:
According to the SOLID principles, specifically the Dependency Inversion Principle, it's better to depend on abstractions rather than concrete classes. Consider using the Storage interface here instead of the concrete StorageImpl class.